### PR TITLE
Lvnod 133 create update event component

### PR DESCRIPTION
**Pull request by yarvanya**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-133)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-134)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-135)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-136)

## Code reviewers

- [ ] @yurii-kokhalevych
- [ ] @denisbodnar
- [ ] @yuliannakorol

### Second Level Review

- [ ] @lhalam
- [ ] @4asof4ik

## Summary of issue

- First, I have created EventsModal component.
- Second, I've craeted EventsModalForm component for EventsModal component.
- Third, I have created the editEventReducer with all needed actions for the EventsModal component.
- Fourth, I've added my child editEventReducer for the main reducer.
- Fifth, I have added simple styles for my Modal window.

#### It will look like this:
- When I click the button "Update", the current event will update and the Modal window close, after that eventDetails components automatially will update itself.

qwqwqw

## CHECK LIST
- [x]  СI passed
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR

### CODE REVIEW

File: app/src/redux/eventReducers.js [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 const FETCH_EVENT_BY_ID_REJECTED = 'FETCH_EVENT_BY_ID_REJECTED';
-import { API } from './../helper/constants';
-import request from './../helper/request';
+const EDIT_EVENT_SUCCESS = 'EDIT_EVENT_SUCCESS';
 
 export const fetchEventById = eventId => {
   return dispatch => {
     return request()
-      .get(API.HOST + API.PORT + '/api/event/' + eventId)
+      .get(API.URL + '/api/event/' + eventId)
```
Comment (4asof4ik) [inline app/src/redux/eventReducers.js]: `${API.URL}/...............................`

File: app/src/redux/editEventReducers.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  };
+}
+
+export function editEvent(event) {
+  return dispatch => {
+    dispatch(editEventRequest());
+    request()
+      .put(API.URL + '/api/event/' + event.id)
+      .send(event)
+      .end((err, res) => {
```
Comment (4asof4ik) [inline app/src/redux/editEventReducers.js]: improve this

File: app/src/containers/Event/EventDetails.js [hunk 3]
```diff
@@ -5,8 +5,10 @@ import { Col, Button, ButtonToolbar, PageHeader, Tabs, Tab, ListGroup, ListGroup
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
 import { GuestsModal } from './../../components';
+import { EventsModal } from './../../components';
```
Comment (4asof4ik) [inline app/src/containers/Event/EventDetails.js]: in line 7 you have imported components

**Review by denisbodnar — DISMISSED** [review]

File: app/src/redux/eventReducers.js [hunk 4]
```diff
@@ -1,13 +1,15 @@
+import { API } from './../helper/constants';
```
Comment (4asof4ik) [inline app/src/redux/eventReducers.js:1]: {API, request }

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

File: app/src/redux/editEventReducers.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  return {
+    type: EDIT_EVENT_REQUEST
+  };
+}
+
+export function editEvent(event) {
+  return dispatch => {
+    dispatch(editEventRequest());
+    request()
+      .put(API.URL + '/api/event/' + event.id)
```
Comment (yuliannakorol) [inline app/src/redux/editEventReducers.js]: use ` ${event.id}`

File: app/src/redux/eventReducers.js (hunk 1 above)
Comment (yuliannakorol) [inline app/src/redux/eventReducers.js]: ${eventId}

File: app/src/redux/eventReducers.js (hunk 4 above)
Comment (yarvanya) [reply app/src/redux/eventReducers.js:1]: It will not be working, while Ura's PR isn't merged.
I will correct it, when it will be done.

**Review by denisbodnar — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

File: app/src/components/EventsModalForm/EventsModalForm.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+);
+
+class EventsModalForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = this.props.event;
+  }
+
+  componentWillMount() {
+    const { event } = this.props;
```
Comment (4asof4ik) [inline app/src/components/EventsModalForm/EventsModalForm.js]: you dont need this const

File: app/src/components/EventsModalForm/EventsModalForm.js [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+  dateTimeFieldHandleChange = date => {
+    this.setState({
+      date_event: date
+    });
+  };
+
+  handleButtonClick = e => {
+    const { closeModal } = this.props;
+    e.preventDefault();
+    this.props.editEventActions.editEvent(this.state);
```
Comment (4asof4ik) [inline app/src/components/EventsModalForm/EventsModalForm.js]: add this into 45 line

**Review by 4asof4ik — APPROVED** [review]
fix and  you can merge

**Pull request merged**